tools/yulPhaser/main.cpp

```diff
@@ -0,0 +1,130 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <tools/yulPhaser/Exceptions.h>
+#include <tools/yulPhaser/Chromosome.h>
+#include <tools/yulPhaser/Exceptions.h>
+#include <tools/yulPhaser/Population.h>
+#include <tools/yulPhaser/Program.h>
+
+#include <libsolutil/CommonIO.h>
+#include <liblangutil/CharStream.h>
+
+#include <boost/filesystem.hpp>
+#include <boost/program_options.hpp>
+
+#include <iostream>
+#include <string>
+
+using namespace std;
+using namespace solidity::langutil;
+using namespace solidity::phaser;
+using namespace solidity::util;
+
+namespace po = boost::program_options;
+
+struct CommandLineParsingResult
+{
+	int exitCode;
+	po::variables_map arguments;
+};
+
+CharStream loadSource(string const& _sourcePath)
```

Comment:
Maybe put inside anonymous namespace?

Reply:
Sure. I did it for other `.cpp` files but it slipped my mind that `main.cpp` is does not get any special treatment from linker in that regard.

Reply:
Done.